File: pastebook-db/pastebook-db/Models/Notification.cs
namespace pastebook_db.Models
{
    public class Notification
    {
        public Guid Id { get; set; }
        public bool HasSeen { get; set; }
        public DateTime NotificationDate { get; set; }
        public string Content { get; set; } = null!;

        // Foreign Key
        // user to receive the notification
        public Guid? UserId { get; set; }
        public virtual User? User { get; set; }

        // Notifications they will receive

        // Foreign Key
        // user to receive the notification
        public Guid? UserRequestId { get; set; }
        public virtual User? UserRequest { get; set; }

        public Guid? PostId { get; set; }
        public virtual Post? Post { get; set; }
        public Guid? AlbumId { get; set; }
        public virtual Album? Album { get; set; }
    }

    public class NotifDTO
    {
        public Guid? Id { get; set; }
        public bool? HasSeen {get; set;}
        public string? NotificationDate { get; set; }
        public string? Content { get; set; }

        //FKs
        public Guid? UserId { get; set; }
        public UserSendDTO? User { get; set; }
        public Guid? PostId { get; set; }
        public Post? Post { get; set; }
        public Guid? AlbumId { get; set; }
        public Post? Album { get; set; }
        public Guid? UserRequestId {  get; set; }
        public UserSendDTO? UserRequest { get; set; }
    }
}

File: postbook-angular/src/app/models/notification.ts
import { Album } from "./album";
import { Post } from "./post";
import { User } from "./user";

export class Notification {
    constructor(
        public id?:string,
        public hasSeen?:boolean,
        public notificationDate?:string,
        public content?:string,
        //FKs
        public userId?:string,
        public user?:User,

        public postId?:string,
        public post?:Post,

        public albumId?:string,
        public album?:Album,

        public userRequestId?:string,
        public userRequest?:User,
    ){}
}
